docs/design/storagenode-graceful-exit.md

```diff
@@ -0,0 +1,62 @@
+# Storage Node Graceful Exit Design Document
+
+
+## OVERVIEW
+
+When a Storage Node wants to leave the network but does not want to lose their escrow we need to have a mechanism for them to exit the network “gracefully”.
+
+This process including the Storage nodes transferring their pieces to other nodes so that the satellite does not have to repair those pieces because of a node exiting abruptly. The process of a Storage Node exiting gracefully including that node requesting a list of Storage Nodes to send their pieces to and updating the satellite with what nodes are now storing those pieces. Graceful Exit for Storage Nodes is beneficially for both the Storage Node and satellite because the storage node receives their escrow and the satellite saves money from not having to repair files.
+
+
+## GOALS
+
+- Give Storage Nodes a mechanism to leave the network while receiving their escrows, and reduce repair caused by node churn on satellites. 
+
+
+## NON GOALS
+- Sending the storage nodes escrows
+	- The sending of tokens will happen through our normal token payment process
+
+
+##SCENARIOS
+
+- A storage node successfully completes a graceful exit
+- A storage node cancels the graceful exit process
+- A storage node shuts down during the graceful exit process
+	- Power loss or reboot 
+- A storage node runs out of bandwidth during the graceful exit process
+- A storage node wants to “partially” exit a satellite
+- A storage node wants to rejoin a satellite it previously exited
```

Comment:
How about a scenario where one or more of the storage nodes that should receive the pieces of the exiting node go offline?

Reply:
I think in that situation the storage node who is exiting the network would need to ask the satellite for another node to upload the data to. I'm not sure what the best course of action would be though, that is only an idea.